source/grid/grid_in.cc

```diff
@@ -2184,483 +2217,495 @@ GridIn<dim, spacedim>::read_msh(std::istream &in)
           in >> line;
         }
 
-      // if the next block is of kind $PartitionedEntities, ignore it
-      if (line == "$PartitionedEntities")
+      // if the next block is of kind $Nodes or $NOD, parse it
+      if (line == begin_nodes_marker[gmsh_file_format == 10 ? 0 : 1])
         {
-          do
+          // now read the nodes list
+          int n_entity_blocks = 1;
```

Comment:
Why redeclare n_entity_blocks that you had declared above? The above declaration is not used, is it necessary?